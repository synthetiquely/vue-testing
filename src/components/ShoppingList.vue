<template>
  <ul class="shopping-list">
    <li
      v-for="item in items"
      :key="item.uuid"
      :class="{ done: item.done }"
    >
      {{ item.text }}
      <button @click="complete(item.uuid)">Done</button>
    </li>
  </ul>
</template>

<script>
export default {
  data() {
    return {
      items: [
        { text: "Beer", done: true, uuid: 1 },
        { text: "Milk", done: false, uuid: 2 },
        { text: "Apple", done: false, uuid: 3 }
      ]
    };
  },
  methods: {
    complete(key) {
      let item = this.items.find(entry => entry.uuid === key);
      item.done = true;
    }
  }
};
</script>
