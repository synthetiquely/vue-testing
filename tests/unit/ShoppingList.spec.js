import Vue from "vue";
import ShoppingList from "@/components/ShoppingList.vue";

describe("Shopping List component", () => {
  let $mounted;

  beforeEach(() => {
    $mounted = new Vue(ShoppingList).$mount();
  });

  it("should match snapshot", () => {
    let $html = $mounted.$el.outerHTML;
    expect($html).toMatchSnapshot();

    $mounted.items[1].done = true;

    Vue.nextTick(() => {
      $html = $mounted.$el.outerHTML;
      expect($html).toMatchSnapshot();
    });
  });

  it("should handle `click` and update `done` attr for a given item", () => {
    let todos = $mounted.$el.querySelectorAll("li");
    let button = todos[1].querySelector("button");
    let customEvent = new Event("click");
    button.dispatchEvent(customEvent);

    expect($mounted.items[1].done).toBeTruthy();
  });
});
