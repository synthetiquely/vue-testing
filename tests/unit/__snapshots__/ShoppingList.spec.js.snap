// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shopping List component should match snapshot 1`] = `
<ul class="shopping-list">
  <li class="done">
    Beer
    <button>Done</button></li>
  <li class="">
    Milk
    <button>Done</button></li>
  <li class="">
    Apple
    <button>Done</button></li>
</ul>
`;

exports[`Shopping List component should match snapshot 2`] = `
<ul class="shopping-list">
  <li class="done">
    Beer
    <button>Done</button></li>
  <li class="done">
    Milk
    <button>Done</button></li>
  <li class="">
    Apple
    <button>Done</button></li>
</ul>
`;
